scripts/get-note-articles.ts

```diff
@@ -0,0 +1,444 @@
+    await git.init();
+
+    // リモートの設定（安全なURL構築）
+    log("Adding GitHub remote...");
+    const remoteUrl = `https://github.com/${owner}/${repoName}.git`;
+    await git.addRemote("origin", remoteUrl);
+
+    // リモートブランチの確認（存在すれば取得）
+    let defaultBranch = "main";
+    try {
+      log("Checking remote repository...");
+      const remoteInfo = await git.listRemote(["--heads", "origin"]);
+
+      if (!remoteInfo) {
+        // リモートリポジトリが空の場合
+        log("Creating new main branch for empty repository...");
+        await git.checkout(["-b", defaultBranch]);
+      } else {
+        // リモートブランチが存在する場合
+        log("Fetching from remote...");
+        await git.fetch("origin");
+
+        // デフォルトブランチの確認
+        defaultBranch = await octokit.repos
+          .get({ owner, repo: repoName })
+          .then((res) => res.data.default_branch)
+          .catch(() => "main");
+
+        log(`Checking out ${defaultBranch} branch...`);
+        try {
+          await git.checkout(["-b", defaultBranch, `origin/${defaultBranch}`]);
+        } catch {
+          // ローカルブランチが既に存在する場合
+          await git.checkout(defaultBranch);
+        }
+      }
+    } catch (error) {
+      throw new Error(
+        `Failed to access remote repository: ${error instanceof Error ? error.message : String(error)}`,
+      );
+    }
+
+    // markdownファイルだけをコピー
+    log("Copying markdown files...");
+    const files = fs.readdirSync(outputDir);
+    let hasChanges = false;
+
+    for (const file of files) {
+      if (file.endsWith(".md")) {
+        const sourcePath = path.join(outputDir, file);
+        const destPath = path.join(tempDir, file);
+        fs.copyFileSync(sourcePath, destPath);
+        hasChanges = true;
+      }
+    }
+
+    if (!hasChanges) {
+      log("No markdown files to push");
+      // 一時ディレクトリを削除
+      fs.rmSync(tempDir, { recursive: true, force: true });
+      return;
+    }
+
+    // 変更のコミットとプッシュ
+    await git.add(".");
+    const status = await git.status();
+
+    if (
+      !status.modified.length &&
+      !status.not_added.length &&
+      !status.created.length
+    ) {
+      log("No changes to push");
+      // 一時ディレクトリを削除
+      fs.rmSync(tempDir, { recursive: true, force: true });
+      return;
+    }
+
+    const timestamp = new Date().toISOString();
+    await git.commit(`Update articles: ${timestamp}`);
+
+    try {
+      log("Pushing to GitHub repository...");
+      await git.push("origin", defaultBranch, ["--set-upstream", "--force"]);
+      log("Successfully pushed changes to GitHub", "success");
+    } catch (error) {
+      throw new Error(
+        `Push failed: ${error instanceof Error ? error.message : String(error)}`,
+      );
+    } finally {
+      // 一時ディレクトリを削除
+      fs.rmSync(tempDir, { recursive: true, force: true });
+    }
+  } catch (error) {
+    throw new Error(
+      `Failed to push to GitHub: ${error instanceof Error ? error.message : String(error)}`,
+    );
+  }
+}
+
+// 新しい関数を追加
```

Comment:
```suggestion
// 記事ファイル名のサニタイズ
```

細かい話だけで失礼ながら、改めてずっと読んできてここが最後引っかかってしまったので念の為コメント差し上げておきます。

※短時間でテスト準備までお疲れ様でした！